src/plugins/mailer/service.ts

```diff
@@ -33,6 +35,74 @@ export class MailerService {
     });
   }
 
+  public translate(lang: string = DEFAULT_LANG) {
+    i18next.changeLanguage(lang);
+    return i18next.t;
+  }
+
+  public async sendEmailChangeRequestConfirmation(
+    oldEmail: string,
+    newEmail: string,
+    userLang: string,
+  ) {
+    const lang = userLang ?? DEFAULT_LANG;
+    const t = this.translate(lang);
+
+    const subject = t(MAIL.CONFIRM_CHANGE_EMAIL_TITLE);
+    const text = t(MAIL.CONFIRM_CHANGE_EMAIL_TEXT, { newEmail });
+
+    const footer = this.buildFooter(lang);
+
+    return this.sendEmail(subject, oldEmail, text, text, footer);
+  }
+
+  public async composeAndSendEmail(
+    email: string,
+    userLang: string,
+    subject: string,
+    buttonText: string,
+    text: string,
+    translationVariables: { [key: string]: string },
+    callToActionLink: string,
+    includeUserAgreement: boolean = false,
+    signUpNotRequested: boolean = false,
```

Comment:
Actually I preferred when the email is built where it will be used, otherwise you have lots of options in your function that are only relevant for one use case. And the day you want a new email format you will need to create a new function just for this email.

Reply:
Any opinion? @ReidyT @spaenleh @Defferrard ?

Reply:
That's a bit the direction I would take with the Builder. You'd build the email where you use it, and then you just pass the mail object to the mailer. What do you think about that @pyphilia? Also, I think if we have to rebuild the same mail in multiple place, we could use a director ([still in the builder pattern](https://refactoring.guru/design-patterns/builder)) to define specific configuration of a mail (specific mail "template").

Reply:
But maybe and advantage with the mail builder, is that the other service don't care how to build the mail, it just have to define what will be in the mail.

Reply:
@pyphilia I found that the problem with constructing the emails all over the place was that eventually everyone just copy-pasted the same code with slight modifications everywhere. This makes sense, since it's an action that you don't write often enough.

Although I admit, I may have gone overboard by trying to centralise e-mail composition. It makes the system a bit too rigid. I think that the solution proposed by @ReidyT could work. With a centralised builder, we can have a guided way to create an e-mail, without it becoming completely rigid.

What do you think ?